docs/specs/meta-clients/web3modal/api.md

```diff
@@ -1,142 +1,255 @@
-# API
+# API (v2)
 
-Web3Modal has it's own api available at [api.web3modal.com](https://api.web3modal.com) that is used to fetch wallet data from a WalletConnect maintained [wallet & dapp registry](https://walletconnect.com/explorer).
+> For the deprecated v1 specification, see [here](./api-v1-legacy.md).
 
-All responses are cached for a minimum of 2h.
+## Query Parameters
 
-## Headers
+All API requests now use query parameters instead of custom headers for SDK metadata, making integration simpler and more compatible with common HTTP clients and browser caching mechanisms.
 
-All api requests must contain following headers. See [types](./types.md)
+### Mandatory Query Parameters
 
-```ts
-interface Headers {
-  "x-project-id": string;
-  "x-sdk-type": SdkType;
-  "x-sdk-version": SdkVersion;
+All API endpoints expect the following mandatory query parameters:
+
+- `st=`: SDK type
+  - Options: `w3m | wcm`
+  - Example: `st=w3m`
+- `sv=`: SDK version
+  - Options: `${framework}-${library}-${version}`
+  - Example: `sv=react-wagmi-4.0.0`
+- `projectId=`: The project ID for tracking and analytics
+  - Options: `string`
+  - Example: `projectId=702e2d45...`
+
+## API Endpoints
+
+### `GET /v2/wallets`
+
+Fetches wallet data based on provided query parameters.
+
+#### Query Parameters
+
+**Required**
+
+- `page`: The page number of results (e.g., `1`)
+- `entries`: Number of entries per page (e.g., `100`)
+
+**Optional**
+
+- `search`: Optional search term (e.g., `MetaMask`)
+- `include`: Optional comma-separated list of wallet IDs to include
+- `exclude`: Optional comma-separated list of wallet IDs to exclude
+- `chains`: Optional comma-separated list of chain identifiers (e.g., `eip155:1`)
+- `platform`: Optional platform filter (`ios` or `android`)
+
+#### Response
+
+```typescript
+{
+  "count": number,
+  "data": [
+    {
+      "id": string,
+      "name": string,
+      "homepage": string,
+      "image_id": string,
+      "order": number,
+      "mobile_link": string | null,
```

Comment:
Recently we had a case were we needed the universal_link only, nowadays the api returns the deeplink by default, and universal_link as a fallback.

Should we consider sending both again?
https://github.com/WalletConnect/web3modal/pull/2352/files

Reply:
Yes definitely 👍  I think the API shouldn't be internalising this logic and the clients should be deciding on which one to use. If we only want to use `deep_link` happy days, but it shouldn't be returning ambiguous values (sometimes deep link sometimes universal) where clients have no choice in the matter.

Reply:
something like this? 

```
"mobile_link": {
   "deep_link": string | null,
    "universal_link": string | null
} | null
```

Reply:
Yes, I think we should mirror what was already the standard in Explorer API, e.g.

```
"mobile": {
  "native": "trust://",
  "universal": "https://link.trustwallet.com"
},
```

One less difference to deal with between the response schemas that way.

Will incorporate into the new PR 👍